Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -95,109 +100,151 @@ namespace AzAssetBrowserRequestHandlerPrivate
         return true;
     }
 
-    void SpawnEntityAtPoint(const ProductAssetBrowserEntry* product, AzQtComponents::ViewportDragContext* viewportDragContext, EntityIdList& spawnList, AzFramework::SliceInstantiationTicket& spawnTicket)
+    void SpawnEntityAtPoint(
+        AZStd::vector<const ProductAssetBrowserEntry*> products,
+        AzQtComponents::ViewportDragContext* viewportDragContext,
+        EntityIdList& spawnList,
+        AzFramework::SliceInstantiationTicket& spawnTicket)
     {
         // Calculate the drop location.
-        if ((!viewportDragContext) || (!product))
+        if ((!viewportDragContext) || products.empty())
         {
             return;
         }
 
+        QWidget* mainWindow = nullptr;
+        EditorRequests::Bus::BroadcastResult(mainWindow, &EditorRequests::GetMainWindow);
+
         const AZ::Transform worldTransform = AZ::Transform::CreateTranslation(viewportDragContext->m_hitLocation);
 
-        // Handle instantiation of slices.
-        if (product->GetAssetType() == AZ::AzTypeInfo<AZ::SliceAsset>::Uuid())
+        struct AssetIdAndComponentTypeId
         {
-            // Instantiate the slice at the specified location.
-            AZ::Data::Asset<AZ::SliceAsset> asset = AZ::Data::AssetManager::Instance().FindOrCreateAsset<AZ::SliceAsset>(product->GetAssetId(), AZ::Data::AssetLoadBehavior::Default);
-            if (asset)
+            AssetIdAndComponentTypeId(AZ::Uuid componentTypeId, AZ::Data::AssetId assetId)
+                : m_componentTypeId(componentTypeId)
+                , m_assetId(assetId)
             {
-                SliceEditorEntityOwnershipServiceRequestBus::BroadcastResult(spawnTicket,
-                    &SliceEditorEntityOwnershipServiceRequests::InstantiateEditorSlice, asset, worldTransform);
             }
-        }
-        else
-        {
-            ScopedUndoBatch undo("Create entities from asset");
+            AZ::Uuid m_componentTypeId = AZ::Uuid::CreateNull();
+            AZ::Data::AssetId m_assetId;
+        };
+        AZStd::vector<AssetIdAndComponentTypeId> assetIdAndComponentTypeId;
+        AZ::ComponentTypeList componentsToAdd;
 
-            //  Add the component(s).
-            AZ::Uuid componentTypeId = AZ::Uuid::CreateNull();
-            AZ::AssetTypeInfoBus::EventResult(componentTypeId, product->GetAssetType(), &AZ::AssetTypeInfo::GetComponentTypeId);
-            if (!componentTypeId.IsNull())
+        for (const AzToolsFramework::AssetBrowser::ProductAssetBrowserEntry* product : products)
+        {
+            // Handle instantiation of slices.
+            if (product->GetAssetType() == AZ::AzTypeInfo<AZ::SliceAsset>::Uuid())
```

Comment:
are we really still supporting SliceAsset?

Reply:
Yeah, until prefabs cover everything slices can do I think